addons/sale_pdf_quote_builder/models/sale_order_template.py

```diff
@@ -1,44 +1,45 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-import base64
-
-from odoo import _, api, fields, models
-
-from odoo.addons.sale_pdf_quote_builder import utils
+from odoo import fields, models
 
 
 class SaleOrderTemplate(models.Model):
     _inherit = 'sale.order.template'
 
-    sale_header = fields.Binary(
-        string="Header pages", default=lambda self: self.env.company.sale_header)
-    sale_header_name = fields.Char(default=lambda self: self.env.company.sale_header_name)
-    sale_footer = fields.Binary(
-        string="Footer pages", default=lambda self: self.env.company.sale_footer)
-    sale_footer_name = fields.Char(default=lambda self: self.env.company.sale_footer_name)
-
-    # === CONSTRAINT METHODS ===#
-
-    @api.constrains('sale_header')
-    def _ensure_header_encryption(self):
+    quotation_document_ids = fields.Many2many(
+        string="Headers and footers",
+        comodel_name='quotation.document',
+        relation='header_footer_quotation_template_rel',
+    )
+    sale_header_ids = fields.Many2many(
+        string="Headers",
+        comodel_name='quotation.document',
+        domain=[('document_type', '=', 'header')],
```

Comment:
maybe we could also filter if the document is already added to the template to not suggest it again?

Reply:
This is a many2many, so it'd be a bit more complicated than that =]